test/e2e/policies.test-suite/topology-aware/n4c16/test06-fuzz/generate.go

```diff
@@ -0,0 +1,128 @@
+}
+
+func createPod(pod string, contcount, cpu, mem int) m.Body {
+	return func(current m.State) m.State {
+		s := current.(*TestState)
+		if s.cpu < cpu*contcount || s.mem < mem*contcount {
+			// refuse from state change if not enough resources
+			return nil
+		}
+		if _, ok := s.podCpuMem[pod]; ok {
+			// refuse to create pod if it is already running
+			return nil
+		}
+		newPodCpuMem := make(map[string][2]int)
+		for k, v := range s.podCpuMem {
+			newPodCpuMem[k] = v
+		}
+		newPodCpuMem[pod] = [2]int{cpu * contcount, mem * contcount}
+		return &TestState{
+			cpu:       s.cpu - cpu*contcount,
+			mem:       s.mem - mem*contcount,
+			podCpuMem: newPodCpuMem,
+		}
+	}
+}
+
+func deletePod(pod string) m.Body {
+	return func(current m.State) m.State {
+		s := current.(*TestState)
+		cpumem, ok := s.podCpuMem[pod]
+		if !ok {
+			// refuse to delete pod if it is not running
+			return nil
+		}
+		newPodCpuMem := make(map[string][2]int)
+		for k, v := range s.podCpuMem {
+			if k != pod {
+				newPodCpuMem[k] = v
+			}
+		}
+		return &TestState{
+			cpu:       s.cpu + cpumem[0],
+			mem:       s.mem + cpumem[1],
+			podCpuMem: newPodCpuMem,
+		}
+	}
+}
+
+func main() {
+	maxMem := 7500         // 7.5GB
+	maxCpu := 15000        // 15 cores
+	maxReservedCpu := 1000 // 1 core
+	maxTestSteps := 3000
+
+	podNames := []string{"gu0", "gu1", "gu2", "gu3", "gu4", "bu0", "bu1", "be0", "be1"}
+
+	model := m.NewModel()
+
+	model.From(func(current m.State) []*m.Transition {
+		s := current.(*TestState)
+		return m.When(s.cpu > 0 && s.mem > 0,
+			m.OnAction("NAME=gu0 CONTCOUNT=1 CPU=200m MEM=1500M create guaranteed").Do(createPod("gu0", 1, 200, 1500)),
+			m.OnAction("NAME=gu1 CONTCOUNT=2 CPU=1000m MEM=500M create guaranteed").Do(createPod("gu1", 2, 1000, 500)),
+			m.OnAction("NAME=gu2 CONTCOUNT=2 CPU=1200m MEM=4500M create guaranteed").Do(createPod("gu2", 2, 1200, 4500)),
+			m.OnAction("NAME=gu3 CONTCOUNT=3 CPU=2000m MEM=500M create guaranteed").Do(createPod("gu3", 3, 2000, 500)),
+			m.OnAction("NAME=gu4 CONTCOUNT=1 CPU=4200m MEM=100M create guaranteed").Do(createPod("gu4", 1, 4200, 100)),
+			m.OnAction("NAME=bu0 CONTCOUNT=1 CPU=1200m MEM=50M CPUREQ=900m MEMREQ=49M CPULIM=1200m MEMLIM=50M create burstable").Do(createPod("bu0", 1, 1200, 50)),
+			m.OnAction("NAME=bu1 CONTCOUNT=2 CPU=1900m MEM=300M CPUREQ=1800m MEMREQ=299M CPULIM=1900m MEMLIM=300M create burstable").Do(createPod("bu1", 2, 1900, 300)),
+			m.OnAction("NAME=be0 CONTCOUNT=1 CPU=0 MEM=0 create besteffort").Do(createPod("be0", 1, 0, 0)),
+			m.OnAction("NAME=be1 CONTCOUNT=3 CPU=0 MEM=0 create besteffort").Do(createPod("be1", 3, 0, 0)))
+	})
+
+	model.From(func(current m.State) []*m.Transition {
+		s := current.(*TestState)
+		ts := []*m.Transition{}
+		for _, pod := range podNames {
+			if _, ok := s.podCpuMem[pod]; ok {
+				ts = append(ts, m.OnAction("NAME=%s kubectl delete pod %s --now", pod, pod).Do(deletePod(pod))...)
```

Comment:
I think this should be `ts = append(ts, m.OnAction("NAME=%s vm-command 'kubectl delete pod %s --now'", pod, pod).Do(deletePod(pod))...)`...

Reply:
Yeah, let's get rid of using confusing `kubectl` function call from shell test lib.

I'm going with vm-command at this phase for minimal blast radius.